### PR TITLE
Tempo: Better fallbacks for metrics query

### LINKED ISSUE #3429: [TraceQL Metrics] Unexpected error when rating by rootName/rootServiceName
image

Reproduced with the same images as here:
https://github.com/grafana/tempo/issues/3428

If I include a filter in the query it works:

image

### PR DESCRIPTION
**Pull request by adrapereira**
**What is this feature?**

This PR fixes a couple of bugs we were seeing when Tempo returned no labels or labels with empty values for a specific query when using the metrics endpoint.

When there are no labels and the results contains a single series, we fallback to the query string, to mimic the Prometheus datasource.

When the value for a label is an empty string we force the series name to be `""` since this is the actual value of the label. In Tempo the values are typed so we know that `""` is the value of empty string instead of the absence of value. 

**Why do we need this feature?**

We were seeing errors and undesired behaviours when querying the metrics endpoint of Tempo.

**Who is this feature for?**

Users of the Tempo datasource. 

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/tempo/issues/3429

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/plugins/datasource/tempo/resultTransformer.ts [hunk 1]
```diff
@@ -633,34 +633,41 @@ function transformToTraceData(data: TraceSearchMetadata) {
...
+        // For single label series, use the label value as the displayName to improve readability
+        name = metricsValueToString(series.labels[0].value);
+      } else {
+        // otherwise build a string using the label keys and values
+        name = `{${series.labels.map((label) => `${label.key}=${metricsValueToString(label.value)}`).join(',')}}`;
+      }
+    }
     return createDataFrame({
-      refId: series.promLabels,
+      refId: `A${index}`,
```
Comment (fabrizio-grafana) [inline public/app/plugins/datasource/tempo/resultTransformer.ts:657]: Why the `A`? 🤔

**Review by fabrizio-grafana — APPROVED** [review]
I didn't test the changes live, but they make sense to me. 

It would be great to add some tests for these, at least as unit tests, for example to avoid the problem with `{}` and `<nil>`, as discussed on Slack.
